relay-server/src/envelope.rs

```diff
@@ -1190,31 +1192,53 @@ impl Envelope {
         }))
     }
 
-    /// Splits the envelope by the given predicate.
-    ///
-    /// The main differents from `split_by()` is this function returns the list of the newly
-    /// constracted envelopes with all the items where the predicate returns `true`. Otherwise it
-    /// returns an empty list.
+    /// Splits the envelope into smaller processible envelopes.
     ///
-    /// The returned envelopes assume the same headers.
-    pub fn split_all_by<F>(&mut self, f: F) -> SmallVec<[Box<Self>; 3]>
-    where
-        F: FnMut(&Item) -> bool,
-    {
-        let mut envelopes = smallvec![];
-        let Some(split_items) = self.split_off_items(|count| count == 0, f) else {
-            return envelopes;
-        };
-
-        let headers = &mut self.headers;
-
-        for item in split_items {
-            // Each item should get an envelope with the new event id.
-            headers.event_id = Some(EventId::new());
+    /// The items which require events are always kept together in one envelope, the reest items
+    /// will be split into separate envelope with one item type each.
+    pub fn into_processible_parts(mut self) -> SmallVec<[Box<Self>; 3]> {
```

Comment:
This function contains queueing / processing business logic that exceeds what should be implemented on a data class like the Envelope. Let's move this function either to queueing (_endpoints/common.rs_) or to the processor.